a10networks/a10-octavia
# [DB-Migration][First Revision]: Added the required new database tables by creating their db models and alembic script.

## Description
Added new database tables required for a10-octavia in a custom database named `a10_octavia` for now, by following the schema diagram(**update_db**) attached in https://a10networks.atlassian.net/browse/STACK-1177, by using the db models and their respective alembic revision script.
Creating the following tables by this:

- amphora_meta
- thunder
- role
- thunder_cluster
- partitions
- project
- ethernet_interface
- trunk_interface
- ve_interface_cluster

## Jira Ticket
https://a10networks.atlassian.net/browse/STACK-1723
https://a10networks.atlassian.net/browse/STACK-1725
https://a10networks.atlassian.net/browse/STACK-1740
https://a10networks.atlassian.net/browse/STACK-1741
https://a10networks.atlassian.net/browse/STACK-1742
https://a10networks.atlassian.net/browse/STACK-1758
https://a10networks.atlassian.net/browse/STACK-1759

## Config
Added changes from : https://github.com/a10networks/a10-octavia/pull/222

## Technical Approach
- Created database models for all the newly required db tables in **models.py**
- Modified the existing VThunder and VRID models as per latest db tables requirements to make it in sync with the revisions.
- Added `target_metadata` to **env.py** for autogenerating the alembic revision script as per modified models.py.
- Autogenerated the alembic revision script **b63ad99c9123_create_required_tables.py** for creating all the newly required tables in database.

## Manual Testing
For now, Tested the table creation on a custom database named a10_octavia.
Step1: Create database a10_octavia:
mysql> **CREATE DATABASE a10_octavia;**

Step2: For connecting to the custom db, add following to [database] section in `/etc/octavia/octavia.conf` file and then restart the octavia-api service:
`connection = mysql+pymysql://root:password@127.0.0.1:3306/a10_octavia`

Step2: From the /path/to/a10-octavia/a10_octavia/db/migration folder run:
$ alembic upgrade head

Step3: Check the newly created tables with their added columns in mysql a10_octavia database
mysql> use a10_octavia;

mysql> show tables;
+-----------------------+
| Tables_in_a10_octavia |
+-----------------------+
| alembic_version |
| amphora_meta |
| ethernet_interface |
| partitions |
| project |
| role |
| thunder |
| thunder_cluster |
| trunk_interface |
| ve_interface_cluster |
| vrid |
| vthunders |
+-----------------------+
